clash-lib/src/Clash/Driver.hs

```diff
@@ -740,6 +755,50 @@ createHDL backend modName seen components domainConfs top topName = flip evalSta
       dom
       domainConfs
 
+writeVerilatorShim
+  :: FilePath
+  -> Id.Identifier
+  -> [(FilePath, ByteString)]
+  -> IO [(FilePath, ByteString)]
+writeVerilatorShim hdlDir topNm filesAndDigests = do
+  let file = Data.Text.unpack (Id.toText topNm) <> "_shim" <.> "cpp"
+  digest <- writeHDL hdlDir (file, pprVerilatorShim topNm)
+  pure ((file, digest) : filesAndDigests)
+
+-- | Create a shim for using verilator, which loads the entity and steps
+-- through simulation until finished.
+--
+pprVerilatorShim :: Id.Identifier -> Doc
+pprVerilatorShim (Id.toText -> topNm) =
+  -- Extra newlines are aggressively inserted so the quasiquoter doesn't wrap
+  -- the outlines lines in the file. It doesn't matter for code inside main,
+  -- but is fatal for the #include directives.
```

Comment:
Perhaps use:

```
import           Data.String.Interpolate      (i)
import           Data.String.Interpolate.Util (unindent)

unindent [i|
```

This quasiquoter was really made for pretty error messages.

Reply:
That didn't work, I'm not going to put any more time into trying to layout this file